[ao][sparsity] Data Sparsifier Benchmarking: Model quality evaluation of the sparsified DLRM model

macandro96 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #81781
* #81780
* __->__ #81779
* #81778

The objective is to perform evaluation of the model quality after sparsifying the embeddings of the dlrm model.
The ```evaluation_model_metrics.py``` makes use of the ```sparse_model_metadata.csv``` file dumped by the
```evaluate_disk_savings.py```. The model metrics such as accuracy, auc, f1 etc are calculated on the test-dataset
for various sparsity levels, block shapes and norms available on the metadata csv file.

**Results**: The model accuracy decreases slowly with sparsity levels. Even at 90% sparsity levels, the model accuracy decreases only by 2%.

After running `evaluate_memory_savings.py`, run: `python evaluate_model_metrics.py --raw_data_file=<path_to_raw_data_txt_file> --processed_data_file=<path_to_kaggleAdDisplayChallenge_processed.npz> --sparse_model_metadata=<path_to_sparse_model_metadata_csv>`

Dependencies: DLRM Repository (https://github.com/facebookresearch/dlrm)

Test Plan: None

Commit:
[ao][sparsity] Data Sparsifier Benchmarking: Model quality evaluation of the sparsified DLRM model

The objective is to perform evaluation of the model quality after sparsifying the embeddings of the dlrm model.
The ```evaluation_model_metrics.py``` makes use of the ```sparse_model_metadata.csv``` file dumped by the
```evaluate_disk_savings.py```. The model metrics such as accuracy, auc, f1 etc are calculated on the test-dataset
for various sparsity levels, block shapes and norms available on the metadata csv file.

Dependencies: DLRM Repository (https://github.com/facebookresearch/dlrm)

Test Plan: None

[ghstack-poisoned]

Commit:
Update on "[ao][sparsity] Data Sparsifier Benchmarking: Model quality evaluation of the sparsified DLRM model"


The objective is to perform evaluation of the model quality after sparsifying the embeddings of the dlrm model.
The ```evaluation_model_metrics.py``` makes use of the ```sparse_model_metadata.csv``` file dumped by the
```evaluate_disk_savings.py```. The model metrics such as accuracy, auc, f1 etc are calculated on the test-dataset
for various sparsity levels, block shapes and norms available on the metadata csv file.

Dependencies: DLRM Repository (https://github.com/facebookresearch/dlrm)

Test Plan: None

[ghstack-poisoned]

Commit:
Update on "[ao][sparsity] Data Sparsifier Benchmarking: Model quality evaluation of the sparsified DLRM model"


The objective is to perform evaluation of the model quality after sparsifying the embeddings of the dlrm model.
The ```evaluation_model_metrics.py``` makes use of the ```sparse_model_metadata.csv``` file dumped by the
```evaluate_disk_savings.py```. The model metrics such as accuracy, auc, f1 etc are calculated on the test-dataset
for various sparsity levels, block shapes and norms available on the metadata csv file.

Dependencies: DLRM Repository (https://github.com/facebookresearch/dlrm)

Test Plan: None

[ghstack-poisoned]

z-a-f reviewed (approved):
Can you add the benchmarking results in the description of this PR before landing? As part of it, also add how the benchmark was run (the CLI command)

Commit:
Update on "[ao][sparsity] Data Sparsifier Benchmarking: Model quality evaluation of the sparsified DLRM model"


The objective is to perform evaluation of the model quality after sparsifying the embeddings of the dlrm model.
The ```evaluation_model_metrics.py``` makes use of the ```sparse_model_metadata.csv``` file dumped by the
```evaluate_disk_savings.py```. The model metrics such as accuracy, auc, f1 etc are calculated on the test-dataset
for various sparsity levels, block shapes and norms available on the metadata csv file.

**Results**: The model accuracy decreases slowly with sparsity levels. Even at 90% sparsity levels, the model accuracy decreases only by 2%.

After running `evaluate_memory_savings.py`, run: `python evaluate_model_metrics.py --raw_data_file=<path_to_raw_data_txt_file> --processed_data_file=<path_to_kaggleAdDisplayChallenge_processed.npz> --sparse_model_metadata=<path_to_sparse_model_metadata_csv>`

Dependencies: DLRM Repository (https://github.com/facebookresearch/dlrm)

Test Plan: None

[ghstack-poisoned]

Commit:
Update on "[ao][sparsity] Data Sparsifier Benchmarking: Model quality evaluation of the sparsified DLRM model"


The objective is to perform evaluation of the model quality after sparsifying the embeddings of the dlrm model.
The ```evaluation_model_metrics.py``` makes use of the ```sparse_model_metadata.csv``` file dumped by the
```evaluate_disk_savings.py```. The model metrics such as accuracy, auc, f1 etc are calculated on the test-dataset
for various sparsity levels, block shapes and norms available on the metadata csv file.

**Results**: The model accuracy decreases slowly with sparsity levels. Even at 90% sparsity levels, the model accuracy decreases only by 2%.

After running `evaluate_memory_savings.py`, run: `python evaluate_model_metrics.py --raw_data_file=<path_to_raw_data_txt_file> --processed_data_file=<path_to_kaggleAdDisplayChallenge_processed.npz> --sparse_model_metadata=<path_to_sparse_model_metadata_csv>`

Dependencies: DLRM Repository (https://github.com/facebookresearch/dlrm)

Test Plan: None

[ghstack-poisoned]

macandro96 commented:
@pytorchbot merge -g
